File: javascript/snippets/getting-started-parameters.mjs
import { parse, bindParams } from 'flyql'
import { generateWhere, newColumn } from 'flyql/generators/clickhouse'

// Parse a query with parameter placeholders
const result = parse("status = $code and env in [$env, 'staging']")

// Bind concrete values to the parameters
bindParams(result.root, { code: 200, env: 'prod' })

// Generate SQL
const columns = {
    status: newColumn("status", false, "Int32", null),
    env: newColumn("env", false, "String", null),
}
const sql = generateWhere(result.root, columns)
console.log(sql)

File: python/snippets/getting_started_parameters.py
from flyql import parse, bind_params
from flyql.generators.clickhouse.generator import to_sql_where
from flyql.generators.clickhouse.column import Column

# Parse a query with parameter placeholders
result = parse("status = $code and env in [$env, 'staging']")

# Bind concrete values to the parameters
bind_params(result.root, {"code": 200, "env": "prod"})

# Generate SQL
columns = {
    "status": Column("status", False, "Int32"),
    "env": Column("env", False, "String"),
}
sql = to_sql_where(result.root, columns)
print(sql)
